sdk/ml/azure-ai-ml/tests/dsl/unittests/test_mldesigner_imports.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+from azure.ai.ml import Input
+from azure.ai.ml.entities import(
+    Component,
+    CommandComponent,
+    PipelineComponent,
+    ValidationResult,
+)
+from azure.ai.ml.dsl._mldesigner import(
+    _AdditionalIncludes,
+    InternalComponent,
+    ErrorTarget,
+)
+
+@pytest.mark.unittest
+@pytest.mark.pipeline_test
+class TestMldesignerImports:
+    """
+    The assertions are NOT SUPPOSED TO BE CHANGED once they are added.
+
+    The attributes are needed for a certain version of mldesigner package, modifying or deleting any of them will cause
+    compatibility issues. If there are new dependencies for mldesigner package, add new assertions in this file.
+    """
+    def test_necessay_attributes(self):
+        assert hasattr(Component, "_customized_validate")
```

Comment:
Shall we also check the interface, like parameter names?

Reply:
Currently these checks are added from real use case in mldesigner, the last time I checked it's component._customized_validate() and pipeline._customized_validate(), no parameters passed in.

Reply:
Oh I don't mean exact this line but all those tests

Reply:
It's good if we can check all those interfaces, but that can also be very complicated and trivial: parameter names like you mentioned, then parameter types, or even inner logic which affect returned values/types, so I guess we'll do some trade-off here, case-by-case.